src/fondant/component/data_io.py

```diff
@@ -135,20 +108,34 @@ def load_dataframe(self) -> dd.DataFrame:
             The Dask dataframe with the field columns in the format (<subset>_<column_name>)
                 as well as the index columns.
         """
-        # load index into dataframe
-        dataframe = self._load_index()
-        for name, subset in self.component_spec.consumes.items():
-            fields = list(subset.fields.keys())
-            subset_df = self._load_subset(name, fields)
-            # left joins -> filter on index
-            dataframe = dd.merge(
-                dataframe,
-                subset_df,
-                left_index=True,
-                right_index=True,
-                how="left",
+        dataframe = None
+        field_mapping = self.manifest.field_mapping
+        for location, fields in field_mapping.items():
```

Comment:
We need to make sure that we read the location containing the latest index first, so the left join below is correct.

Reply:
I've updated the `field_mapping` method. The method returns an immutable OrderedDict now. The first entry is the location of the dataset of the latest index.

Reply:
I might have missed something but it seems to be that we're currently loading/merging all the fields defined in the manifest since that's what `field_mapping` contains. Shouldn't we only base ourselves on the fields defined in the `consumes` section of the component spec?

Reply:
Good catch :sweat_smile:

I think this should work:
```python
fields = [field for field in fields if 
          (field in self.component_spec.consumes or field == DEFAULT_INDEX_NAME)]
```

Reply:
I've started the implementation of the consumes and produces in #665.
Maybe it is easier to include the changes of the component package there here as well! I'll take a look. 

Basically, the idea was the following:

- Load the dataframe based on the manifest, later on renaming the columns based on the defined consumes before the dataset is passed to the transform method. 
- When produce is defined renaming the columns of the output dataframe accordingly to the produce mapping, write it to the bucket and use the written column names for the output manifest. 

Am I overlooking something?

Reply:
> Good catch 😅
> 
> I think this should work:
> 
> ```python
> fields = [field for field in fields if 
>           (field in self.component_spec.consumes or field == DEFAULT_INDEX_NAME)]
> ```

I don't see why we need the manifest field mapping in the first place. 

Can't we do something like this (a bit similar to how it was before):

```
for field_name in self.component_spec.consumes.items():
             field_location = self.manifest.fields[field_name].location
```